hbogolib/handlereu.py

```diff
@@ -869,8 +876,92 @@ def categories(self):
+            self.login()
+
+        jsonrsp = self.get_from_hbogo(self.API_GET_BOOKMARK+self.GOcustomerId+'/ROM/1')
+
+        try:
+            if jsonrsp[0]['ErrorMessage']:
+                self.log("List Error: " + str(jsonrsp[0]['ErrorMessage']))
+                xbmcgui.Dialog().ok(self.LB_ERROR, jsonrsp[0]['ErrorMessage'])
+        except KeyError:
+            pass  # all is ok no error message just pass
+        except Exception:
+            self.log("Unexpected error: " + traceback.format_exc())
+     
+        for container_index in range(0, str(jsonrsp).count("contentType")):
+            jsonrsp2 = self.get_from_hbogo(self.API_GET_BY_ID + '/ENG/' + self.API_PLATFORM + '/' + str(jsonrsp[container_index]['externalId']) 
+                                                              + '/' + str(jsonrsp[container_index]['contentType']))
+            self.addLink(jsonrsp2, HbogoConstants.ACTION_PLAY)
+        KodiUtil.endDir(self.handle, self.use_content_type)
+
+    def get_Last_10_History(self, url):
+        if not self.chk_login():
+            self.login()
+        self.log("List: " + str(url))
+
+        if not self.chk_login():
+            self.login()
+
+        jsonrsp = self.get_from_hbogo(self.API_GET_HISTORY+self.GOcustomerId+'/ROM/1')
+
+        try:
+            if jsonrsp[0]['ErrorMessage']:
+                self.log("List Error: " + str(jsonrsp[0]['ErrorMessage']))
+                xbmcgui.Dialog().ok(self.LB_ERROR, jsonrsp[0]['ErrorMessage'])
+        except KeyError:
+            pass  # all is ok no error message just pass
+        except Exception:
+            self.log("Unexpected error: " + traceback.format_exc())
+     
+        for container_index in range(0, min(10, str(jsonrsp).count("contentType"))):
+            jsonrsp2 = self.get_from_hbogo(self.API_GET_BY_ID + '/ENG/' + self.API_PLATFORM + '/' + str(jsonrsp[container_index]['externalId']) 
+                                                              + '/' + str(jsonrsp[container_index]['contentType']))
+            self.addLink(jsonrsp2, HbogoConstants.ACTION_PLAY)
+        KodiUtil.endDir(self.handle, self.use_content_type)
+
+
+    def get_Watching_History(self, url):
+        if not self.chk_login():
+            self.login()
+        self.log("List: " + str(url))
+
+        if not self.chk_login():
+            self.login()
+
+        jsonrsp = self.get_from_hbogo(self.API_GET_HISTORY+self.GOcustomerId+'/ROM/3')
+
+        try:
+            if jsonrsp[0]['ErrorMessage']:
+                self.log("List Error: " + str(jsonrsp[0]['ErrorMessage']))
+                xbmcgui.Dialog().ok(self.LB_ERROR, jsonrsp[0]['ErrorMessage'])
+        except KeyError:
+            pass  # all is ok no error message just pass
+        except Exception:
+            self.log("Unexpected error: " + traceback.format_exc())
+
+        for container_index in range(0, str(jsonrsp).count("contentType")):
+            container_item = jsonrsp[container_index]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'container_item' is assigned to but never used](https://app.codacy.com/app/arvvoid/plugin.video.hbogoeu/pullRequest?prid=4593966)